### PR TITLE
DBPW - Update sdk to get new package name

### PR DESCRIPTION
**Pull request by pcman312**
# Overview
Updates the sdk package to get the new `dbplugin` package name

# Design of Change
`go mod` and some import renames

# Related Issues/Pull Requests
https://github.com/hashicorp/vault/pull/10151

# Test Output

<details>

<summary>Test Output</summary>
</details>

# Contributor Checklist
- [x] Docs not applicable here
- [x] Add output for any tests not ran in CI to the PR description (eg, acceptance tests)
- [x] Backwards compatible with dbplugin version 5

### CODE REVIEW

File: mongodbatlas.go [hunk 1]
```diff
@@ -7,25 +7,25 @@ import (
...
-	"github.com/hashicorp/vault/sdk/database/newdbplugin"
 	"github.com/mitchellh/mapstructure"
 	"github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
 )
 
 const mongoDBAtlasTypeName = "mongodbatlas"
 
 // Verify interface is implemented
-var _ newdbplugin.Database = &MongoDBAtlas{}
+var _ dbplugin.Database = &MongoDBAtlas{}
```
Comment (tvoran) [inline mongodbatlas.go:20]:
```suggestion
var _ dbplugin.Database = (*MongoDBAtlas)(nil)
```

File: go.sum [hunk 2]
```diff
@@ -183,11 +183,12 @@ github.com/hashicorp/golang-lru v0.5.3 h1:YPkqC67at8FYaadspW/6uE0COsBxS2656RLEr8
 github.com/hashicorp/golang-lru v0.5.3/go.mod h1:iADmTwqILo4mZ8BN3D2Q6+9jd8WM5uGBxy+E8yxSoD4=
 github.com/hashicorp/hcl v1.0.0 h1:0Anlzjpi4vEasTeNFn2mLJgTSwt0+6sfsiTG8qcWGx4=
 github.com/hashicorp/hcl v1.0.0/go.mod h1:E5yfLk+7swimpb2L/Alb/PJmXilQ/rhwaUYs4T20WEQ=
+github.com/hashicorp/vault v1.5.4 h1:yOEWTliHNsNMCPtop61CRCAWTQbSzwGgzEdWACpt1NY=
```
Comment (tvoran) [inline go.sum:186]: ?

File: vendor/modules.txt [hunk 3]
```diff
@@ -72,13 +72,14 @@ github.com/hashicorp/hcl/json/scanner
...
+# github.com/hashicorp/vault/sdk v0.1.14-0.20201015192012-a69ee0f65a28
 github.com/hashicorp/vault/sdk/database/dbplugin
+github.com/hashicorp/vault/sdk/database/dbplugin/v5
+github.com/hashicorp/vault/sdk/database/dbplugin/v5/proto
 github.com/hashicorp/vault/sdk/database/helper/connutil
 github.com/hashicorp/vault/sdk/database/helper/credsutil
 github.com/hashicorp/vault/sdk/database/helper/dbutil
 github.com/hashicorp/vault/sdk/database/newdbplugin
-github.com/hashicorp/vault/sdk/database/newdbplugin/proto
 github.com/hashicorp/vault/sdk/database/newdbplugin/testing
```
Comment (tvoran) [inline vendor/modules.txt:83]: `go mod tidy && go mod vendor` gets rid of these for me.

File: mongodbatlas.go (hunk 1 above)
Comment (pcman312) [reply mongodbatlas.go]: Done

File: go.sum (hunk 2 above)
Comment (pcman312) [reply go.sum]: Removed

File: vendor/modules.txt (hunk 3 above)
Comment (pcman312) [reply vendor/modules.txt]: Done

**Review by tvoran — APPROVED** [review]

**Review by calvn — APPROVED** [review]
Acceptance tests passed:
```
=== RUN   TestIntegrationDatabaseUser_Initialize
--- PASS: TestIntegrationDatabaseUser_Initialize (0.00s)
=== RUN   TestAcceptanceDatabaseUser_CreateUser
    mongodbatlas_test.go:189: Asserting username: v-test-9am2ysfrgciwJ
--- PASS: TestAcceptanceDatabaseUser_CreateUser (28.09s)
=== RUN   TestAcceptanceDatabaseUser_CreateUserWithSpecialChar
    mongodbatlas_test.go:234: Asserting username: v-test-7my0NbEyvhxIB
--- PASS: TestAcceptanceDatabaseUser_CreateUserWithSpecialChar (32.07s)
=== RUN   TestAcceptanceDatabaseUser_DeleteUser
    mongodbatlas_test.go:283: Asserting username: v-test-gFTDEflDzI4He
--- PASS: TestAcceptanceDatabaseUser_DeleteUser (61.67s)
=== RUN   TestAcceptanceDatabaseUser_UpdateUser_Password
    mongodbatlas_test.go:327: Asserting username: testmongouser
    mongodbatlas_test.go:340: Asserting username: testmongouser
--- PASS: TestAcceptanceDatabaseUser_UpdateUser_Password (62.97s)
PASS

```

**Pull request merged**